src/connection.ts

```diff
@@ -1129,28 +1137,42 @@ export class Connection extends EventEmitter {
         return
       }
 
+      // Find the smallest packet amount we tried in case we ran into Txx errors
+      const smallestPacketAmount = testPacketAmounts.reduce((min: any, amount: any) => BigNumber.min(min, new BigNumber(amount)), new BigNumber(Infinity))
       // If we get here the first volley failed, try new volley using all unique packet amounts based on the max packets
       testPacketAmounts = maxPacketAmounts
         .filter((amount: any) => !amount.isEqualTo(new BigNumber(Infinity)))
         .reduce((acc: any, curr: any) => [...new Set([...acc, curr.toString()])], [])
 
+      // Check for any Txx Errors
+      if (packetErrors.some((code: string) => code[0] === 'T')) {
+        const timeLeft = DEFAULT_PACKET_TIMEOUT - retryDelay - (Date.now() - startTime)
+        if (timeLeft > 0) {
+          const reducedPacketAmount = smallestPacketAmount.minus(smallestPacketAmount.dividedToIntegerBy(3))
+          this.log.debug(`got Txx error(s), waiting ${retryDelay}ms and reducing packet amount to ${reducedPacketAmount} before sending another test packet`)
+          testPacketAmounts = [...testPacketAmounts, reducedPacketAmount]
+          await new Promise((resolve, reject) => setTimeout(resolve, retryDelay))
+          retryDelay *= RETRY_DELAY_INCREASE_FACTOR
+        } else {
+          this.log.debug(`got Txx error but not trying again`)
```

Comment:
Since all of this is in a while loop, won't it keep retrying anyway, just without the delay?

Reply:
Good point, we don't really need the timeLeft logic here at all since there is a max of 20 and when no testPacketAmounts are found the loop will end. Removed the else statement and the timeleft logic so it just does the backoff and reduction in the case of Txx errors.